cargo-concordium/src/build.rs

```diff
@@ -148,36 +386,100 @@ pub fn build_contract(
         }
     };
 
-    let metadata = get_crate_metadata(cargo_args)?;
-
-    let target_dir = format!("{}/concordium", metadata.target_directory);
+    let (metadata, args_without_manifest) = get_crate_metadata(cargo_args)?;
 
     let package = metadata
         .root_package()
         .context("Unable to determine package.")?;
 
-    let result = Command::new("cargo")
-        .arg("build")
-        .args(["--target", "wasm32-unknown-unknown"])
-        .args(["--release"])
-        .args(["--target-dir", target_dir.as_str()])
-        .args(cargo_args)
-        .stdout(Stdio::inherit())
-        .stderr(Stdio::inherit())
-        .output()
-        .context("Could not use cargo build.")?;
+    let package_root = package
+        .manifest_path
+        .parent()
+        .context("Unable to get package root path.")?;
+
+    let wasm_file_name = format!("{}.wasm", to_snake_case(package.name.as_str()));
+
+    // TODO: cargo-concordium should check if wasm32 target is installed
```

Comment:
Don't need to worry about that

Reply:
Then the todo should be removed 😊

Reply:
I want to add a task to do this instead. It's not part of this feature, it would be nice improvement to cargo-concordium though.

Reply:
Added https://github.com/Concordium/concordium-smart-contract-tools/issues/126